internal/astutils.go

```diff
@@ -7,3 +7,27 @@ import (
 func IsFieldEmbedded(field *ast.Field) bool {
 	return len(field.Names) == 0
 }
+
+func GetFieldSyncMutex(field *ast.Field) (*ast.SelectorExpr, bool) {
```

Comment:
You don't need to factorize.
You can move this code inside the function `checkForbiddenEmbeddedField`.